### PR TITLE
feat(attendance_check): punishment

### PR DESCRIPTION
**Pull request by R1verrrr**
I asked Kien and he said it should be still in the modelsm not FE

### CODE REVIEW

File: models/attendance_check.py [hunk 1]
```diff
@@ -4,10 +4,13 @@
 from pydantic import BaseModel, Field
 
 if sys.version_info >= (3, 11):
-    from typing import Self
+    from typing import Self, TYPE_CHECKING
 else:
     from typing_extensions import Self
 
+if TYPE.CHECKING:
```
Comment (j1nxie) [inline models/attendance_check.py:11]: should be `TYPE_CHECKING`

File: models/attendance_check.py [hunk 2]
```diff
@@ -35,12 +38,20 @@ def set_start_date(self, start_date: datetime) -> Result[Self, str]:
         self.start_date = start_date.strftime("%Y-%m-%d")
         return Ok(self)
 
-    def add_attendance(self, date: datetime, is_present: bool) -> Result[Self, str]:
+    def add_attendance(self, date: datetime, is_present: bool, payroll: Payroll) -> Result[Self, str]:
```
Comment (j1nxie) [inline models/attendance_check.py:41]: the code being here makes sense too, as long as it works

i forgot we had this method for the backend

**Review by j1nxie — CHANGES_REQUESTED** [review]

File: models/attendance_check.py (hunk 1 above)
Comment (R1verrrr) [reply models/attendance_check.py:11]: okii
sorry my mistake

File: models/attendance_check.py [hunk 3]
```diff
@@ -4,10 +4,13 @@
 from pydantic import BaseModel, Field
 
 if sys.version_info >= (3, 11):
-    from typing import Self
+    from typing import Self, TYPE_CHECKING
 else:
     from typing_extensions import Self
```
Comment (j1nxie) [inline models/attendance_check.py:9]: also you should import `TYPE_CHECKING` on this line too, just so python <= 3.10 doesn't break.

**Review by j1nxie — CHANGES_REQUESTED** [review]

File: models/attendance_check.py (hunk 3 above)
Comment (R1verrrr) [reply models/attendance_check.py]: I fixed it

File: models/attendance_check.py [hunk 4]
```diff
@@ -6,9 +6,9 @@
 if sys.version_info >= (3, 11):
     from typing import Self, TYPE_CHECKING
 else:
-    from typing_extensions import Self
+    from typing_extensions import Self, TYPE_CEHCKING
```
Comment (j1nxie) [inline models/attendance_check.py:9]: misspell.

**Review by j1nxie — CHANGES_REQUESTED** [review]

**Review by j1nxie — APPROVED** [review]

**Comment by j1nxie** [comment]
lgtm

**Pull request merged**